### DISCUSSION TITLE
Change Authorization in the header before request

### DISCUSSION THREAD

**Discussion by SofianeHamlaoui** [pallets/flask#5339] [Q&A] [answered]
Hello, 

So to explain the situation in the backend I'm receiving the JWT token encrypted !

What I want to do is to decrypt the token, replace the headers['Authorization'] with the decrypted JWT token on the before_request decorator, but I couldn't do that I tried changing it but it's not changing at all. 

Any ideas how can I do that ?

**Comment by ThiefMaster** [comment]
That seems like a strange thing to do. Why not decode it in a `before_request` handler and store the results e.g. in `g.jwt_data`.

A HTTP header is always a string, but decoded JWT is usually a JSON object. That does not belong into `request.headers`.

**Reply by ThiefMaster** [reply]
Post some real code. I have the feeling you're doing something wrong...

**Reply by ThiefMaster** [reply, marked as answer]
`==` is comparison, not assignment.

Beyond that, I'm not going to help you changing the header value since I strongly believe that you're doing the wrong thing. Store your decoded JWT elsewhere.
